### DISCUSSION TITLE
Check diff when using actions/Checkout

### DISCUSSION THREAD

**Discussion by DaHuoKolmostar** [community/community#26173] [Actions]
<p>I want to konw the diff comparing to the last commit in a Github Actions workflow.</p>
<p>With Checkout@v1, I can obtain the whole repo, therefore I can call “git diff HEAD^” to check the diff. However, it downloads the whole repo, which takes much longer time when dealing with a repo with a long history.</p>
<p>With Checkout@v2, it only downloads the current commit, and “.git” directory is not downloaded, therefore “git diff HEAD^” can not execute.</p>
<p>I wonder if there is an easy way to checkout the diff.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>When use <a href="https://github.com/actions/checkout#checkout-v2" rel="nofollow noopener">Checkout@v2</a>, only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set fetch-depth to fetch more history. In your case, you could set fetch-depth=2. Then you could run </p>
<p>git diff HEAD^ HEAD to get the diff comparing to the last commit. </p>
<p>Please see my example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181095926-978323fd-a19f-4c9b-a077-11ec8c3d9979.png" alt="fetch depth.png" data-base62-sha1="wfICSqCijV9qYSAVaGw1poYcV29" width="291" height="290"></p>
